src/modules/Game.class.js

```diff
@@ -1,68 +1,265 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  static Status = {
+    idle: 'idle',
+    playing: 'playing',
+    win: 'win',
+    lose: 'lose',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.state = initialState.map((row) => [...row]);
+    this.status = Game.Status.idle;
+    this.initialState = initialState;
+    this.score = 0;
+  }
+
+  moveLeft() {
+    if (this.status === Game.Status.playing) {
+      let isMovable = false;
+
+      for (let r = 0; r < 4; r++) {
+        const values = [];
+
+        for (let c = 0; c < 4; c++) {
+          if (this.state[r][c] !== 0) {
+            values.push(this.state[r][c]);
+          }
+        }
+
+        for (let i = 0; i < values.length; i++) {
+          if (values[i] === values[i + 1]) {
+            values[i] *= 2;
+            values[i + 1] = 0;
+            this.score += values[i];
+            isMovable = true;
+          }
+        }
+
+        const updatedRow = values.filter((value) => value !== 0);
+
+        while (updatedRow.length < 4) {
+          updatedRow.push(0);
+        }
+
+        for (let c = 0; c < 4; c++) {
+          if (this.state[r][c] !== updatedRow[c]) {
+            this.state[r][c] = updatedRow[c];
+            isMovable = true;
+          }
+        }
+      }
+
+      if (isMovable) {
+        this.getRandomCells();
+        this.checkGameStatus();
+      }
+    }
+  }
+  moveRight() {
+    if (this.status === Game.Status.playing) {
+      let isMovable = false;
+
+      for (let r = 0; r < 4; r++) {
+        const values = [];
+
+        for (let c = 3; c >= 0; c--) {
+          if (this.state[r][c] !== 0) {
+            values.push(this.state[r][c]);
+          }
+        }
+
+        for (let i = 0; i < values.length; i++) {
+          if (values[i] === values[i + 1]) {
+            values[i] *= 2;
+            values[i + 1] = 0;
+            this.score += values[i];
+            isMovable = true;
+          }
```

Comment:
The same potential bug as in `moveLeft` method is present here. Ensure that after merging, the next cell is skipped to avoid multiple merges in one move.